src/apps/nagle_rcv.c

```diff
@@ -152,31 +153,30 @@ main(int argc, char *argv[])
 	}
 	
 	/* Set some basic values which depend on the address family. */
-#if TEST_V6
-        hst = gethostbyname2(local_host, AF_INET6);
-        if (hst == NULL || hst->h_length < 1) {
-                fprintf(stderr, "%s: bad hostname: %s\n", argv[0], local_host);
+        if (strcmp(local_host, "0") == 0)
+                local_host = "0.0.0.0";
+
+        memset(port_buffer, 0, sizeof(port_buffer));
```

Comment:
This line is not needed, snprintf will put the \0 in the end safely already:
       The  functions  snprintf()  and  vsnprintf()  write  at most size bytes
       (including the terminating null byte ('\0')) to str.